src/test/java/seedu/address/ui/testutil/GuiTestAssert.java

```diff
@@ -34,8 +37,69 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
         assertEquals(expectedPerson.getPhone().value, actualCard.getPhone());
         assertEquals(expectedPerson.getEmail().value, actualCard.getEmail());
         assertEquals(expectedPerson.getAddress().value, actualCard.getAddress());
-        assertEquals(expectedPerson.getTags().stream().map(tag -> tag.tagName).collect(Collectors.toList()),
-                actualCard.getTags());
+        assertTagEquals(expectedPerson, actualCard);
+    }
+
+    /**
+     * Asserts that{@code actualCard} displays the same tag and colour of {@code expectedPerson}
+     * @param expectedPerson
+     * @param actualCard
+     */
+    private static void assertTagEquals(Person expectedPerson, PersonCardHandle actualCard) {
+        List<String> expectedTags = expectedPerson.getTags()
+                .stream().map(tag -> tag.tagName)
+                .collect(Collectors.toList());
+
+        assertEquals(expectedTags, actualCard.getTags());
+
+        expectedTags.forEach(tag -> assertEquals(Arrays
+                .asList(LABEL_STYLE, getTagColorStyleFor(tag)), actualCard.getTagStyleClasses(tag)));
+    }
+
+    /**
+     *TODO: Update in future if more colours are included
+     */
+    private static String getTagColorStyleFor(String tagName) {
+
+        switch(tagName) {
+
+        case "owesMoney":
+            return "red";
+
+        case "friends":
+            return "red";
+
+        case "friend":
+            return "orange";
+
+        case "husband":
+            return "orange";
+
+        case "colleagues":
+            return "green";
+
+        case "classmates":
+            return "red";
+
+        case "neighbours":
+            return "green";
+
+        case "family":
+            return "orange";
+
+        case "urgent":
+            return "red";
+
+        case "Urgent":
+            return "red";
+
+        case "UrGEnT":
```

Comment:
??? Why so many variations of the word `urgent`?

Reply:
Will update